tests/sqlparser_postgres.rs

```diff
@@ -3363,6 +3387,68 @@ fn test_fn_arg_with_value_operator() {
     }
 }
 
+#[test]
+fn json_object_with_null_clause() {
+    match pg().verified_expr("JSON_OBJECT('name' VALUE 'value' NULL ON NULL)") {
+        Expr::Function(Function { args: FunctionArguments::List(FunctionArgumentList { args, clauses, .. }), .. }) => {
+            assert!(matches!(
+                &args[..],
+                &[FunctionArg::ExprNamed { operator: FunctionArgOperator::Value, .. }]
+            ), "Invalid function argument: {args:?}");
+            assert_eq!(
+                clauses,
+                vec![FunctionArgumentClause::JsonNullClause(JsonNullClause::NullOnNull)],
+                "Expected: NULL ON NULL to be parsed as a null treatment clause, but got {clauses:?}"
+            );
+        }
+        other => panic!("Expected: JSON_OBJECT('name' VALUE 'value' NULL ON NULL) to be parsed as a function, but got {other:?}"),
+    }
+}
+
+#[test]
+fn json_object_with_returning_clause() {
+    match pg().verified_expr("JSON_OBJECT('name' VALUE 'value' RETURNING JSONB)") {
+        Expr::Function(Function { args: FunctionArguments::List(FunctionArgumentList { args, clauses, .. }), .. }) => {
+            assert!(matches!(
+                &args[..],
+                &[FunctionArg::ExprNamed { operator: FunctionArgOperator::Value, .. }]
+            ), "Invalid function argument: {args:?}");
+            assert_eq!(
+                clauses,
+                vec![FunctionArgumentClause::JsonReturningClause(JsonReturningClause {
+                    data_type: DataType::JSONB
+                })],
+                "Expected: RETURNING JSONB to be parsed as a returning clause, but got {clauses:?}"
+            );
+        }
+        other => panic!("Expected: JSON_OBJECT('name' VALUE 'value' RETURNING jsonb) to be parsed as a function, but got {other:?}"),
+    }
+}
+
+#[test]
+fn json_object_only_returning_clause() {
+    match pg().verified_expr("JSON_OBJECT(RETURNING JSONB)") {
+        Expr::Function(Function {
+            args: FunctionArguments::List(FunctionArgumentList { args, clauses, .. }),
+            ..
+        }) => {
+            assert!(args.is_empty(), "Expected no arguments, got: {args:?}");
+            assert_eq!(
+                clauses,
+                vec![FunctionArgumentClause::JsonReturningClause(
+                    JsonReturningClause {
+                        data_type: DataType::JSONB
+                    }
+                )],
+                "Expected: RETURNING JSONB to be parsed as a returning clause, but got {clauses:?}"
+            );
+        }
+        other => panic!(
+            "Expected: JSON_OBJECT(RETURNING jsonb) to be parsed as a function, but got {other:?}"
+        ),
+    }
+}
```

Comment:
since they're covering the same feature, can we merge these test cases into one `parse_json_object()` test function?